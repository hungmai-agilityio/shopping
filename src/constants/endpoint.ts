export const END_POINT = {
  HOME: '/',
  SHOP: '/shop',
  ABOUT_US: '/about',
  CONTACT: '/contact',
  SIGN_IN: '/signIn',
  SIGN_UP: '/signUp',
  WISHLIST: '/wishlist',
  PRODUCTS: '/products',
  CATEGORY: '/category',
  COMMENT: '/comment',
  USERS: '/users',
  PROFILE: '/profiles',
  CART: '/cart',
  CHECKOUT: '/checkout'
};
