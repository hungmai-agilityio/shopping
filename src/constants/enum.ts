export enum TYPE {
  PRIMARY = 'primary',
  SECOND = 'second',
  THIRD = 'third',
  FILE = 'file',
  TEXT = 'text',
  PASSWORD = 'password',
  EMAIL = 'email',
  COMMENT = 'comment',
  CONTACT = 'contact'
}

export enum SIZE {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large'
}

export enum TAG {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6',
  P = 'p',
  SPAN = 'span'
}

export enum FONT_SIZE {
  LARGE = 'text-heading',
  MEDIUM = 'text-4xl',
  SMALL = 'text-2xl',
  X_SMALL = 'lg:text-lg text-sm',
  MIMI = 'text-sm'
}

export enum FONT_WEIGHT {
  THIN = 'thin',
  LIGHT = 'light',
  NORMAL = 'normal',
  MEDIUM = 'medium',
  SEMIBOLD = 'semibold',
  BOLD = 'bold'
}

export enum CELL {
  TH = 'th',
  TD = 'td'
}

export enum CURRENT {
  USER = 'user',
  PROFILE = 'profile'
}

export enum GENDER {
  MALE = '0',
  FEMALE = '1',
  OTHER = '2'
}

export enum KEY {
  USER = 'currentUser'
}

export enum STATUS {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning'
}

export enum TIME_STATUS {
  STALE_TIME = 1000 * 60 * 60, // milliseconds * seconds * minutes = 1 hour
  COOKIE_TIME = 60 * 60 * 24 * 2 // seconds * minutes * hour * day
}