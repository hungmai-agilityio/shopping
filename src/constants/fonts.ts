import { Mada, Poppins, Roboto, Signika } from 'next/font/google';

export const mada = Mada({
  weight: ['400', '700'],
  subsets: ['latin']
});

export const popping = Poppins({
  subsets: ['latin'],
  weight: ['400', '700']
})

export const roboto = Roboto({
  subsets: ['latin'],
  weight: ['400', '700']
})

export const signika = Signika({
  subsets: ['latin'],
  weight: ['400', '700']
})