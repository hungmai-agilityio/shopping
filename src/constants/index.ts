export * from '@/constants/fonts';
export * from '@/constants/enum';
export * from '@/constants/endpoint';
export * from '@/constants/messages';
export * from '@/constants/list';
