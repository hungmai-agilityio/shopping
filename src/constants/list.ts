import { END_POINT } from "@/constants/endpoint";

export const VALID_IMAGE = [
  'image/png',
  'image/jpeg',
  'image/jpg',
];

export const slideProducts = [
  {
    id: 'i7g6i9i2-8f89-6h15-9i0e-6j2f7j33g6i6',
    name: 'Basic T-Shirt',
    price: 15,
    description: 'A comfortable and stylish basic t-shirt.',
    category: 't-shirts',
    colors: ['White', 'Blue', 'Green'],
    sizes: ['S', 'M', 'L'],
    image: 'https://i.ibb.co/njkCQHq/t-shirt-3.webp',
    imageDetail: [
      'https://i.ibb.co/njkCQHq/t-shirt-3.webp',
      'https://i.ibb.co/njkCQHq/t-shirt-3.webp',
      'https://i.ibb.co/njkCQHq/t-shirt-3.webp'
    ],
    rating: '10k',
    sell: 'Best quality',
    stoct: 1710
  },
  {
    id: 'u3k8h9f1-9g22-3h67-2g13-5k6g9k45g4t3',
    name: 'Sporty T-Shirt',
    price: 25,
    description: 'Perfect for your workout sessions.',
    category: 't-shirts',
    colors: ['Black', 'Gray', 'white', 'Maroon'],
    sizes: ['M', 'L', 'XL'],
    image: 'https://i.ibb.co/WtgcmWR/t-shirt-2.webp',
    imageDetail: [
      'https://i.ibb.co/WtgcmWR/t-shirt-2.webp',
      'https://i.ibb.co/WtgcmWR/t-shirt-2.webp',
      'https://i.ibb.co/WtgcmWR/t-shirt-2.webp'
    ],
    rating: '1k',
    sell: 'Best quality',
    stoct: 1118
  },
  {
    id: 'y4l7i8o1-2j33-4k56-8p92-7m9b8j45h7u8',
    name: 'Casual T-Shirt',
    price: 20,
    description: 'Stay casual and comfortable.',
    category: 't-shirts',
    colors: ['White', 'Yellow', 'Pink'],
    sizes: ['XS', 'S', 'M', 'L'],
    image: 'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp',
    imageDetail: [
      'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp',
      'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp',
      'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp'
    ],
    rating: '2k',
    sell: 'Best quality',
    stoct: 1710
  },
  {
    id: 'o9b5l2e3-5t78-2v45-7h92-4c8h7d99a3u1',
    name: 'Graphic T-Shirt',
    price: 24,
    description: 'Unique graphic design for a standout look.',
    category: 't-shirts',
    colors: ['Blue', 'Purple'],
    sizes: ['S', 'M', 'L', 'XL'],
    image: 'https://i.ibb.co/fvhz0L9/t-shirt-1.webp',
    imageDetail: [
      'https://i.ibb.co/fvhz0L9/t-shirt-1.webp',
      'https://i.ibb.co/fvhz0L9/t-shirt-1.webp'
    ],
    rating: '9k',
    sell: 'Best quality',
    stoct: 1710
  }
];

export const slideImage = [
  'https://i.ibb.co/MRTgRYj/shirt-1.webp',
  'https://i.ibb.co/h1pRWGY/jacket-5-2.webp',
  'https://i.ibb.co/3d7b9SB/jean-2.webp'
];

export const navItems = [
  { name: 'Home', url: END_POINT.HOME },
  { name: 'About Us', url: END_POINT.ABOUT_US },
  { name: 'Contact', url: END_POINT.CONTACT }
];

export const itemsHome = [
  { name: 'Product', url: END_POINT.HOME },
  { name: 'Categories', url: END_POINT.HOME },
  { name: 'Login', url: END_POINT.SIGN_IN }
];

export const itemsShop = [
  { name: 'T-Shirt', url: END_POINT.HOME },
  { name: 'Jacket', url: END_POINT.HOME },
  { name: 'Shirt', url: END_POINT.HOME },
  { name: 'Jeans', url: END_POINT.HOME }
];

export const itemsCategories = [
  { name: 'Men', url: END_POINT.HOME },
  { name: 'Children', url: END_POINT.HOME },
  { name: 'Woman', url: END_POINT.HOME }
];

export const itemsContact = [{ name: 'mangcoding@gmail.com', url: '' }];