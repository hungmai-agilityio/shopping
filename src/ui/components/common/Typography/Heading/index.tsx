import { FONT_WEIGHT, mada, TAG } from '@/constants';
import { Typography } from '@/ui/components';
import { ReactNode } from 'react';

interface HeadingProp {
  color?: string;
  children: ReactNode;
}
export const Heading = ({ color = 'text-white', children }: HeadingProp) => {
  return (
    <Typography
      fontWeight={FONT_WEIGHT.BOLD}
      color={color}
      tag={TAG.H1}
      className={`${mada.className} capitalize lg:text-heading text-xl my-5 leading-normal`}
    >
      {children}
    </Typography>
  );
};
