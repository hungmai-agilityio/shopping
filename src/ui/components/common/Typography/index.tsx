import { ReactNode } from 'react';
import clsx from 'clsx';

// Constants
import { FONT_SIZE, FONT_WEIGHT, TAG } from '@/constants';

interface TypographyProps {
  children: ReactNode;
  className?: string;
  color?: string;
  fontWeight?: FONT_WEIGHT;
  size?: FONT_SIZE;
  tag?: TAG;
}

const Typography = ({
  children,
  tag = TAG.P,
  fontWeight = FONT_WEIGHT.NORMAL,
  color = 'text-black',
  size = FONT_SIZE.SMALL,
  className
}: TypographyProps) => {
  const Typo = tag;

  return (
    <Typo className={clsx(`font-${fontWeight}`, color, size, className)}>
      {children}
    </Typo>
  );
};

export default Typography;
