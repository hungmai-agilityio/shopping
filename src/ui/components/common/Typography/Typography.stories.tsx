import { Meta, StoryObj } from '@storybook/react';

// Constant
import { FONT_SIZE, FONT_WEIGHT, TAG } from '@/constants';

//  Component
import Typography from '@/ui/components/common/Typography';

const meta: Meta<typeof Typography> = {
  component: Typography,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    tag: {
      description: 'HTML Tag for the Typography component'
    },
    fontWeight: {
      description: 'Font weight for the text'
    },
    color: {
      description: 'Text color class (ex: `text-red-500`, `text-blue-700`)'
    },
    size: {
      description: 'Predefined text size based on the given constants'
    },
    children: {
      description: 'Content of the Typography component'
    }
  }
};

export default meta;
type Story = StoryObj<typeof Typography>;

export const Default: Story = {
  args: {
    children: 'This is a default paragraph text.',
    tag: TAG.P,
    fontWeight: FONT_WEIGHT.NORMAL,
    color: 'text-black',
    size: FONT_SIZE.SMALL
  }
};

export const Heading: Story = {
  args: {
    children: 'This is a Heading',
    tag: TAG.H1,
    fontWeight: FONT_WEIGHT.BOLD,
    color: 'text-purple-500',
    size: FONT_SIZE.LARGE
  }
};

export const Title: Story = {
  args: {
    children: 'This is a Title',
    tag: TAG.H3,
    fontWeight: FONT_WEIGHT.BOLD,
    color: 'text-green-500',
    size: FONT_SIZE.MEDIUM
  }
};

export const CustomText: Story = {
  args: {
    children: 'This is a custom styled text.',
    tag: TAG.SPAN,
    fontWeight: FONT_WEIGHT.LIGHT,
    color: 'text-blue-600',
    size: FONT_SIZE.X_SMALL,
    className: 'italic underline'
  }
};
