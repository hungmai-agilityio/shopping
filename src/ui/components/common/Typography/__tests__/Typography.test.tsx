// Constants
import { FONT_SIZE, FONT_WEIGHT, TAG } from '@/constants';
import Typography from '@/ui/components/common/Typography';
import { render, screen } from '@testing-library/react';

describe('Typography component', () => {
  const defaultProps = {
    children: 'This is a default paragraph text.',
    tag: TAG.P,
    fontWeight: FONT_WEIGHT.NORMAL,
    color: 'text-black',
    size: FONT_SIZE.SMALL
  };

  it('render with default props', () => {
    render(<Typography {...defaultProps} />);
    const typo = screen.getByText('This is a default paragraph text.');
    expect(typo).toBeInTheDocument();
    expect(typo).toHaveClass('font-normal text-black');
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<Typography {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
