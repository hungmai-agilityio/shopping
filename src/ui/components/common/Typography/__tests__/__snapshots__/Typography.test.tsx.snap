// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component should be render match to snapshot 1`] = `
<div>
  <p
    class="font-normal text-black text-2xl"
  >
    This is a default paragraph text.
  </p>
</div>
`;
